### PR TITLE
Reimplement release creation workflow

### PR DESCRIPTION
**Pull request by kwankyu**
Fixes the issue raised in https://github.com/sagemath/website/pull/480#issuecomment-2516392023

I manually edited the release https://github.com/sagemath/sage/releases/tag/10.5 generated by an workflow implemented in https://github.com/sagemath/website/pull/480 to correct the changelog https://github.com/sagemath/website/blob/master/src/changelogs/sage-10.5.txt, which is now in good shape. But see, for example, https://github.com/sagemath/sage/releases/tag/10.4 that contains all changes in betas and rcs.

This PR is for automatic generation of a release that contains only changes from the last release.

test: 
https://github.com/kwankyu/sage/releases
https://github.com/kwankyu/sage/actions/runs/12191414258/job/34010494299 

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Comment by kwankyu** [comment]
@soham30rane Would you review?

File: .github/workflows/dist.yml [hunk 1]
```diff
@@ -119,10 +119,27 @@ jobs:
...
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer $GITHUB_TOKEN" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/${{ github.repository }}/releases \
+            -d '{
+              "tag_name": "${{ github.ref_name }}",
+              "previous_tag_name": "${latest_release_tag}",
+              "prerelease": ${{ contains(github.ref, 'beta') || contains(github.ref, 'rc') }},
+              "generate_release_notes": true
+            }'
```
Comment (soham30rane) [inline .github/workflows/dist.yml:139]: I went through the [official docs](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release) and didn't found parameter `previous_tag_name` for **Create a release** endpoint. So it seems, even if `generate_release_notes` is set to true, the `previous_tag_name` plays no role.

The parameter `previous_tag_name` exists for [generate release notes](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#generate-release-notes-content-for-a-release) endpoint. This endpoint will return the `name` and `body` according to specified parameters, which could be then used to create a release using [create a release](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release) endpoint

Comment (kwankyu) [reply .github/workflows/dist.yml:139]: Nonetheless it works fine.

It seems that `"generate_release_notes": true` internally triggers the "generate release notes" endpoint as well, which uses `previous_tag_name`.

Comment (kwankyu) [reply .github/workflows/dist.yml:139]: I will check again.

**Comment by kwankyu** [comment]
Sorry for confusing state of this PR.

By an accident related with my privileged account (damn it!), I erroneously merged this PR branch to the develop branch. I instantly restored the develop branch, but I cannot restore this PR to the previous state.

So I created a new PR #39194 for the branch. Sorry for inconveniences.

**Pull request merged**